src/steps/05-PerformanceRequirements/DOW/ComputeForm.spec.ts

```diff
@@ -0,0 +1,700 @@
+        "classification": "U",
+        "sys_tags": "",
+        "sys_created_by": "person-ctr@ccpo.mil"
+      }
+  ]
+
+  const availablePeriodCheckboxItems =  [
+    {
+      "id": "BASE",
+      "label": "Base period",
+      "value": "base_01"
+    },
+    {
+      "id": "OPTION1",
+      "label": "Option period 1",
+      "value": "option_01"
+    }
+  ]
+
+  const regionCheckboxOptions = [
+    {
+      id: "CONUSEast",
+      label: "CONUS East",
+      value: "CONUS East",
+    },
+    {
+      id: "CONUSCentral",
+      label: "CONUS Central",
+      value: "CONUS Central",
+    },
+    {
+      id: "CONUSWest",
+      label: "CONUS West",
+      value: "CONUS West",
+    },
+    {
+      id: "OCONUS",
+      label: "OCONUS",
+      value: "OCONUS",
+    },
+    {
+      id: "Other",
+      label: "Other",
+      value: "OtherRegion",
+    }];
+  
+    const periodDTO = [
+      [
+        {
+          "period_unit": "YEAR",
+          "period_unit_count": "1",
+          "period_type": "BASE",
+          "option_order": "1"
+        },
+          {
+          "period_unit": "YEAR",
+          "period_unit_count": "1",
+          "period_type": "BASE",
+          "option_order": "2"
+        },
+        {
+          "period_unit": "YEAR",
+          "period_unit_count": "1",
+          "period_type": "BASE",
+          "option_order": "3"
+        }
+      ]
+    ]
+
+  beforeEach(() => {
+    vuetify = new Vuetify();
+    wrapper = mount(ComputeForm, {
+      localVue,
+      vuetify,
+      mocks:{
+        $store:{
+          DescriptionOfWork:{
+            computeObject: computeData,
+          }
+        },
+        
+      },
+      propsData: {
+        computeData: computeData
+      }
+    });
+  });
+
+
+  describe("Initialization....", () => {
+    it("tests that component renders successfully", async () => {
+      expect(wrapper.exists()).toBe(true);
+    });
+  });
+
+  describe("Vue Lifecyle Hook Testing...", ()=>{
+    it("mounted", ()=>{
+      expect(wrapper.vm.$data.formHasBeenTouched).toBe(false);
+    })
+  })
+
+  describe("Method Testing...", ()=>{
+    it ("loadOnEnter() - is to set data.firstTimeHere to true ", async()=>{
+      const spy = await jest.spyOn(wrapper.vm, "loadOnEnter");
+      wrapper.vm.loadOnEnter();
+      expect(wrapper.vm.$data.firstTimeHere).toBe(true);
+    });
+
+    it("@Watch errorBagChange() - sets data.formHasErrors when errors occur ", async()=>{
+      // set the initial data.errorBagValues
+      wrapper.vm.$data.errorBagValues = [false, false];
+      const computeForm = wrapper.findComponent({ref: "computeForm"});
+      computeForm.vm.$data.errorBag = {810:true, 827: false}
+
+      // change the data.errorBagValues value to activate data reactivity
+      await wrapper.setData({
+        errorBagValues: [true, false]
+      })
+      expect(await wrapper.vm.$data.formHasErrors).toBe(true);
+    });
+  
+    it("validate() - ensures validate() has been called", async ()=>{
```

Comment:
This sets up a spy on a method, calls that method and then makes sure the method was called. So we're really testing our test, not testing our code. Is there some other behavior that we can test here? If not we're better off without anything here.

Reply:
yeah, that should have been deleted.  Will remove it...

Reply:
Removed